sdk/python/feast/repo_config.py

```diff
@@ -190,6 +194,37 @@ def _validate_offline_store_config(cls, values):
 
         return values
 
+    @root_validator(pre=True)
+    def _validate_feature_server_config(cls, values):
+        # Having no feature server is the default.
+        if "feature_server" not in values:
+            return values
+
+        # Make sure that the provider configuration is set. We need it to set the defaults
+        assert "provider" in values
+
+        # Make sure that the type is not set, since we will set it based on the provider.
+        assert "type" not in values
```

Comment:
should we be checking type in values["feature_server"] instead?

Reply:
Also can you replace asserts with if/raise checks? (similar to @woop 's comment)